src/app/loginComponents/accounts/internal/change-username/change-username.component.ts

```diff
@@ -80,11 +80,7 @@ export class ChangeUsernameComponent implements OnInit {
       )
       .subscribe(
         (userExists: boolean) => {
-          if (userExists && this.username === this.user.username) {
-            this.usernameTaken = false;
-          } else {
-            this.usernameTaken = userExists;
-          }
+          this.usernameTaken = userExists;
```

Comment:
Does anyone know why the old code was written the way it was?  The new code looks correct, but whenever I encounter something "weird" in the codebase, I wonder if there is a reason...

Reply:
I'm guessing it was written the previous way because _technically_ if a user is looking to change their username their _own_ username is still an "option" so we didn't mark it as taken because they own it/can use it. But it throws an error if they try to click save on their present username so it doesn't make sense to allow it as an option